src/main/webapp/app/exercises/shared/participation-submission/participation-submission.component.ts

```diff
@@ -72,4 +129,52 @@ export class ParticipationSubmissionComponent implements OnInit {
                 }
             });
     }
+
+    getName() {
+        if (this.participation?.type === ParticipationType.STUDENT) {
+            return (this.participation as StudentParticipation).student?.name || (this.participation as StudentParticipation).team?.name;
+        } else if (this.participation?.type === ParticipationType.SOLUTION) {
+            return this.translate.instant('artemisApp.participation.solutionParticipation');
+        } else if (this.participation?.type === ParticipationType.TEMPLATE) {
+            return this.translate.instant('artemisApp.participation.templateParticipation');
+        }
+        return 'N/A';
+    }
+
+    isTemplateOrSolutionParticipation(): boolean {
```

Comment:
This function actually return `boolean | undefined`. I think if you want to return boolean than you need to change `this.participation?....` to `!this.participation?...`

Reply:
no it doesn't. If `this.participation?...` is undefined the `===` results into false, because undefined is not equal to  ParticipationType.TEMPLATE